src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -19418,3 +19418,146 @@ def timedelta_property(
+            The keyword arguments to groupby().
+        agg_args: Sequence
+            Positional arguments to the unique() aggregation function. This
+            parameter must be empty because unique() does not take positional
+            arguments, but we keep the parameter to match the interface of
+            Modin's BaseQueryCompiler.
+        agg_kwargs: dict
+            Keyword arguments to the unique() aggregation function. This
+            parameter must be empty because unique() does not take keyword
+            arguments, but we keep the parameter to match the interface of
+            Modin's BaseQueryCompiler.
+        numeric_only: bool
+            This parameter is meaningless as unique() does not take a
+            numeric_only parameter, but we keep the parameter to match the
+            interface of Modin's BaseQueryCompiler.
+        is_series_groupby: bool
+            Whether this method is called via SeriesGroupBy as opposed to
+            DataFrameGroupBy. This parameter should always be true, but we keep
+            it to match the interface of Modin's BaseQueryCompiler.
+        drop: bool, default False
+            Whether the `by` columns are internal this dataframe.
+
+        Returns
+        -------
+        A new SnowflakeQueryCompiler with the unique values of the singular
+        data column for each group.
+        """
+        assert axis == 0, "Internal error. SeriesGroupBy.unique() axis should be 0."
+        assert len(agg_args) == 0, (
+            "Internal error. SeriesGroupBy.unique() does not take "
+            + "aggregation arguments."
+        )
+        assert len(agg_kwargs) == 0, (
+            "Internal error. SeriesGroupBy.unique() does not take "
+            + "aggregation arguments."
+        )
+        assert (
+            is_series_groupby is True
+        ), "Internal error. Only SeriesGroupBy has a unique() method."
+
+        compiler = SnowflakeQueryCompiler(
+            self._modin_frame.ensure_row_position_column()
+        )
+        by_list = extract_groupby_column_pandas_labels(
+            compiler, by, groupby_kwargs.get("level", None)
+        )
+        by_snowflake_quoted_identifiers_list = []
+        for (
+            entry
+        ) in compiler._modin_frame.get_snowflake_quoted_identifiers_group_by_pandas_labels(
+            by_list
+        ):
+            assert len(entry) == 1, (
+                "Internal error. Each grouping label should correspond to a "
+                + "single Snowpark column."
+            )
+            by_snowflake_quoted_identifiers_list.append(entry[0])
+
+        # There is no built-in snowflake function to aggregation unique values
+        # of a column into array while preserving a certain order. We implement
+        # the aggregation in the following steps:
+        # 1) Project a new column representing the row position of each row
+        #    within each combination of group + data column value.
```

Comment:
Why we need group + data columns here? Should it be just grouping columns?

Reply:
We need to keep the rows where the combination of group + data column value appears for the next time. I explained that in the next step. Let me know if I can make the wording clearer.

Reply:
Ah, I understand it now. No this is fine. Thanks!